XenAdmin/Dialogs/VMProtectionRecovery/VMProtectionPoliciesDialog.cs

```diff
@@ -201,43 +201,54 @@ private void LoadPolicies()
             dataGridView1.SuspendLayout();
             var selectedPolicy = currentSelected;
             dataGridView1.Rows.Clear();
+	    var policyList = VMGroup<T>.VMPolicies(Pool.Connection.Cache);
+
+            /* creating a dictionary to hold (policy_uuid, message list) */
+
+            Dictionary<string, List<XenAPI.Message>> policyMessage = new Dictionary<string, List<XenAPI.Message>>();
+    
+            /* populate the dictionary with policy uuid */
+
+            foreach (var policy in policyList)
+            {
+                policy.PolicyAlerts.Clear();
+                List<XenAPI.Message> messageList = new List<XenAPI.Message>();
+                policyMessage.Add(policy.uuid, messageList);
+            }
+       
+            /* iterate through all messages and populate the dictionary with message list */

```

Comment:
We should try to reuse the code in GetVMSSAlertsAction to populate the PolicyAlerts for each polocy with the last 10 messages.